pkg/rke2/np.go

```diff
@@ -57,14 +57,15 @@ func setNetworkPolicy(ctx context.Context, namespace string, cs *kubernetes.Clie
 	if _, ok := ns.Annotations[namespaceAnnotationNetworkPolicy]; !ok {
 		if _, err := cs.NetworkingV1().NetworkPolicies(namespace).Get(ctx, defaultNetworkPolicyName, metav1.GetOptions{}); err != nil {
 			if !apierrors.IsNotFound(err) {
+				if err := cs.NetworkingV1().NetworkPolicies(namespace).Delete(ctx, defaultNetworkPolicyName, metav1.DeleteOptions{}); err != nil {
```

Comment:
The last two sentences on the description for setNetworkPolicy have me scratching my head:
> If the annoation doesnt' exist, we check to see if the policy exists. If it does, we delete it, and create the new default policy.

The intent is to always replace it when a server starts up? Seems like that will result in unnecessary churn in a multi-server setup.

As for the execution of that logic, this branch to delete/create will never be visited because the `cs.NetworkingV1().NetworkPolicies(namespace).Get()` call will not return an error if the policy exists.

Reply:
Marking as resolved: worked through this with @briandowns on a zoom.